modules/farm_fd2/src/entrypoints/transplanting/lib.submit.unit.cy.js

```diff
@@ -0,0 +1,221 @@
+import { lib } from './lib.js';
+import { lib as traySeedingLib } from '../tray_seeding/lib.js';
+import * as farmosUtil from '@libs/farmosUtil/farmosUtil';
+
+describe('Submission using the transplanting lib.', () => {
+  let traySeedingForm = {
+    seedingDate: '1950-01-02',
+    cropName: 'BROCCOLI',
+    locationName: 'CHUAU',
+    trays: 25,
+    traySize: '200',
+    seedsPerCell: 3,
+    comment: 'A comment',
+  };
+
+  let form = {
+    cropName: 'BROCCOLI',
+    picked: [],
+    transplantingDate: '1950-01-02',
+    location: 'ALF',
+    beds: ['ALF-1', 'ALF-3'],
+    bedFeet: 100,
+    bedWidth: 60,
+    rowsPerBed: '1',
+    equipment: ['Tractor'],
+    depth: 6,
+    speed: 5,
+    comment: 'A comment',
+  };
+
+  let cropToTerm = null;
+  let result = null;
+  let fieldMap = null;
+  let bedMap = null;
+
+  before(() => {
+    cy.restoreLocalStorage();
+    cy.restoreSessionStorage();
+
+    cy.wrap(farmosUtil.getFieldIdToAssetMap()).then((map) => {
+      fieldMap = map;
+    });
+
+    cy.wrap(farmosUtil.getBedIdToAssetMap()).then((map) => {
+      bedMap = map;
+    });
+
+    cy.wrap(farmosUtil.getCropNameToTermMap()).then((map) => {
+      cropToTerm = map;
+    });
+
+    cy.wrap(traySeedingLib.submitForm(traySeedingForm), { timeout: 10000 })
+      .then(() => {
+        return cy.wrap(farmosUtil.getSeedlings(traySeedingForm.cropName), {
+          timeout: 10000,
+        });
+      })
+      .then((res) => {
+        form.picked[0] = { trays: 25, data: res[res.length - 1] };
```

Comment:
Better to use `traySeedingForm.trays` here instead of hard coded `25` in case the form changes.